crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -3136,6 +3150,7 @@ where
                 if self.is_stub {
                     if self.enabled(Rule::DuplicateUnionMember)
                         && self.semantic.in_type_definition()
```

Comment:
We may want to add these (`&& self.semantic.in_type_definition()`) to the `Rule::UnnecessaryLiteralUnion` condition too. I can't quite remember why they're necessary here. I'm guessing it's because for `|`, if it's used outside of a type definition, it's not a union operator (it's a bitwise or, I think).

Reply:
That's true for the `DuplicateUnionMember` rule where they're checking for _any_ duplicate items in a `|` but here we know we are constrained to literal types where this does apply.

We test for this at 

https://github.com/charliermarsh/ruff/blob/1759d914456b62dfc479518ce8717949935f6c87/crates/ruff/resources/test/fixtures/flake8_pyi/PYI030.pyi#L28

You can see that a bitwise or of literals resolve to a `Union` outside of an annotation:

```
In [3]: Literal[1] | Literal[2]
Out[3]: typing.Union[typing.Literal[1], typing.Literal[2]]

In [4]: x = Literal[1] | Literal[2]

In [5]: x
Out[5]: typing.Union[typing.Literal[1], typing.Literal[2]]
```

Reply:
I guess it's possible that someone has `Literal[1] | bar | Literal[2]` where `bar` implements some sort of custom bitwise or behavior and this rule no longer holds but I'm not sure that's worth handling

Reply:
Interesting, that makes sense. Thanks for the thorough follow-up!